cincinnati/src/lib.rs

```diff
@@ -92,10 +98,13 @@ impl Graph {
         let release = release.into();
         match self.find_by_version(&release.version()) {
             Some(id) => {
-                let mut node = self.dag.node_weight_mut(id.0).unwrap();
+                let mut node = match self.dag.node_weight_mut(id.0) {
+                    Some(n) => n,
+                    None => bail!("failed to retrieve node weight"),
```

Comment:
This can never happen since `find_by_version` has given us a valid node ID. I'd rather we just fix `find_by_version` to return a mutable node weight (`Release`) so we don't have to do the lookup again.

Reply:
Playing around with this, I wasn't able to get this to work. It seems to be a limitation in petgraph. I'd rather we just leave the panic behavior in place, since it does indicate a bug in the code.